Scripts/AwsEC2GetPublicSGRules/AwsEC2GetPublicSGRules.py

```diff
@@ -0,0 +1,138 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+import json
+import copy
+
+
+def get_dict_value(data, key):
+    """ Returns dict value for a given key (case insensitive) """
+    for key_name in data.keys():
+        if key_name.lower() == key.lower():
+            return data[key_name]
+
+    return None
+
+
+def raise_error(error):
+    return {
+        'Type': entryTypes['error'],
+        'ContentsFormat': formats['text'],
+        'Contents': str(error)
+    }
+
+
+def get_ec2_sg_public_rules(group_id, ip_permissions, checked_protocol=None, checked_from_port=None,
+                            checked_to_port=None, region=None, include_ipv6='no'):
+    """
+        Get the list of public
+        which can be passed on to the following command:
+        aws-ec2-revoke-security-group-ingress-rule
+    """
+    public_rules = []
+    for rule in ip_permissions:
+        # Check protocol
+        protocol = get_dict_value(rule, 'IpProtocol')
+        if protocol != '-1':
+            if checked_protocol.lower() != protocol.lower():
+                continue
+
+        bad_rule = {
+            'groupId': group_id,
+            'ipProtocol': protocol
+        }
+
+        if region:
+            bad_rule.update(region=region)
+
+        # Check the ports
+        from_port = get_dict_value(rule, 'FromPort')
+        to_port = get_dict_value(rule, 'ToPort')
+        if from_port and to_port:
```

Comment:
what if only one is specified?

Reply:
AWS IP permissions rule will always have one of the following pattern:
- for `All Traffic`, ipProtocol will be `-1` and there won't be `fromPort` nor `toPort`. 
- for `All TCP`, `All UDP`, or any other regular rules, AWS will always have `fromPort` and `toPort` in the rule definition

For that reason, the output will either include both fromPort-toPort in the rule or none